lms/djangoapps/grades/api/tests/test_views.py

```diff
@@ -0,0 +1,259 @@
+"""
+Tests for the views
+"""
+from datetime import datetime
+
+from django.core.urlresolvers import reverse
+from mock import patch
+from opaque_keys import InvalidKeyError
+from pytz import UTC
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from student.tests.factories import CourseEnrollmentFactory, UserFactory
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase, TEST_DATA_SPLIT_MODULESTORE
+
+
+class CurrentGradeViewTest(SharedModuleStoreTestCase, APITestCase):
+    """
+    Tests for the Current Grade View
+
+    The following tests assume that the grading policy is the edX default one:
+    {
+        "GRADER": [
+            {
+                "drop_count": 2,
+                "min_count": 12,
+                "short_label": "HW",
+                "type": "Homework",
+                "weight": 0.15
+            },
+            {
+                "drop_count": 2,
+                "min_count": 12,
+                "type": "Lab",
+                "weight": 0.15
+            },
+            {
+                "drop_count": 0,
+                "min_count": 1,
+                "short_label": "Midterm",
+                "type": "Midterm Exam",
+                "weight": 0.3
+            },
+            {
+                "drop_count": 0,
+                "min_count": 1,
+                "short_label": "Final",
+                "type": "Final Exam",
+                "weight": 0.4
+            }
+        ],
+        "GRADE_CUTOFFS": {
+            "Pass": 0.5
+        }
+    }
+    """
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    @classmethod
+    def setUpClass(cls):
+        super(CurrentGradeViewTest, cls).setUpClass()
+
+        cls.course = CourseFactory.create(display_name='test course', run="Testing_course")
+
+        chapter = ItemFactory.create(
+            category='chapter',
+            parent_location=cls.course.location,
+            display_name="Chapter 1",
+        )
+        # creating a problem for each type and minimum count needed by the grading policy
+        # min_count means that that section is not considered if the student does not answer at least
+        # "min_count" problems
```

Comment:
Nit: These 2 lines -> `A section is not considered if the student answers less than "min_count" problems`